crates/ty_server/src/lib.rs

```diff
@@ -14,7 +15,7 @@ mod server;
 mod session;
 mod system;
 
-#[cfg(test)]
+#[cfg(feature = "testing")]
 pub mod test;
```

Comment:
I would move this implementation into the `test` folder too. See how I shared the comment `CliTest` infrastructure in `ty/tests/cli`

Reply:
I've pushed this change. I've moved the tests under `e2e/` directory and moved the `TestServer` and related data structures to `e2e/main.rs` file.